frontend/src/components/signup.tsx

```diff
@@ -12,13 +12,23 @@ export const SignUp = () => {
     const [lastName, setLastName] = useState<string>('');
     const [email, setEmail] = useState<string>('');
     const [password, setPassword] = useState<string>('');
+    const [validLength, setValidLength] = useState<boolean>(false); 
+    const [hasNumber, setHasNumber] = useState<boolean>(false); 
+    const [specialChar, setSpecialChar] = useState<boolean>(false); 
```

Comment:
generally good rule of thumb for variables that are meant to hold boolean vals is to make the var name as if it's based on the question.

`isvalidLength` vs `validLength`
`hasNumber` is good!
`hasSpecialChar` vs `specialChar`

something like `specialChar` is a little ambiguous. is it a string var holding a special char? does it transform special chars in a string? etc.